pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1185,6 +1185,11 @@ class Timedelta(_Timedelta):
                 "Value must be Timedelta, string, integer, "
                 "float, timedelta or convertible")
 
+        if unit in {'Y', 'y', 'M'}:
```

Comment:
One more small comment: I think we should put this check above the block, as `unit` is already used in there / results are already returned. I think it is better to do this deprecation check before that.